lib/puppet/functions.rb

```diff
@@ -603,9 +620,29 @@ def initialize(loader, name)
     # @api public
     #
     def type(assignment_string)
-      result = parser.parse_string("type #{assignment_string}", nil) # no file source :-(
+      # Get location to use in case of error - this produces ruby filename and where call to 'type' occurred
+      # but strips off the rest of the internal "where" as it is not meaningful to user.
+      #
+      rb_location = caller[0].gsub(/:in.*$/, '')
```

Comment:
Seems wasteful to resolve this regular expression for cases when the parse is successful. Can you move it into the rescue block?